// Copyright 2021 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package nearby.sharing.proto;

import "sharing/proto/rpc_resources.proto";

option optimize_for = LITE_RUNTIME;

// Request to list public certificate objects.
message ListPublicCertificatesRequest {
  // Required. The resource name determines which public certificates to list.
  // The special prefix "users/me" lists the requesters own share targets. This
  // is of the format "users/*/devices/*".
  string parent = 1;

  // Optional limit on the number of ShareTarget objects to check for
  // PublicCertificates for the response. Further PublicCertificates items may
  // be obtained by including the page_token in a subsequent request. If this is
  //  not set or zero, a reasonable default value is used.
  int32 page_size = 2;

  // Optional pagination token, returned earlier via
  // [ListPublicCertificatesResponse.next_page_token]
  string page_token = 3;

  // Optional. Represents certificates already available on local device.
  repeated bytes secret_ids = 4;
}

// Response that contains the public certificates available to calling device.
message ListPublicCertificatesResponse {
  // Optional. A token to retrieve the next page of results when used in
  // [ListPublicCertificatesRequest].
  string next_page_token = 1;

  // Optional. Public certificates allowed to be accessed by the calling local
  // device.
  repeated PublicCertificate public_certificates = 2;
}
